components/places/src/ffi.rs

```diff
@@ -143,9 +144,41 @@ implement_into_ffi_by_protobuf!(msg_types::HistoryVisitInfos);
 implement_into_ffi_by_protobuf!(msg_types::HistoryVisitInfosWithBound);
 implement_into_ffi_by_protobuf!(msg_types::BookmarkNode);
 implement_into_ffi_by_protobuf!(msg_types::BookmarkNodeList);
-implement_into_ffi_by_protobuf!(msg_types::HistoryMetadata);
-implement_into_ffi_by_protobuf!(msg_types::HistoryMetadataList);
 implement_into_ffi_by_delegation!(
     crate::storage::bookmarks::PublicNode,
     msg_types::BookmarkNode
 );
+
+// XXX - this should be in some other crate?
+// uniffi!
```

Comment:
I think it would be reasonable to include this in UniFFI. One thing to watch out for would be that UniFFI depends on `ffi-support` and exposes the depended-on version as `uniffi::deps::ffi_support`. If you want to move this macro into UniFFI, it may be worth doing `use uniffi::deps::ffi_support` rather than taking our own separate dependency on it, in order to ensure that we use the same version.

Reply:
I actually backed away from that, and replaced that comment with:
```
/// This is only necessary because we have a kinda "hybrid" FFI situation -
/// some things generated via UniFFI, others by hand. Because this macro only
/// makes sense in this Frankenstein world it's not in UniFFI itself.
```